engines/dfc_provider/config/routes.rb

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 DfcProvider::Engine.routes.draw do
+  resources :addresses, only: [:show]
   resources :enterprises, only: [:show] do
     resources :catalog_items, only: [:index, :show, :update]
     resources :supplied_products, only: [:create, :show, :update]
```

Comment:
We could have nested this under enterprises again to just support enterprise addresses. That would have made the authorisation check easier. But then we would need add more controllers for all kinds of addresses in the future, like order shipping address or invoice billing address, customer default shipping address etc.

So to simplify the API, we have only one endpoint for the general Address data type and search for associations to authorise access. All knowledge about address associations is now in one place.